src/evaluators/MemberExpression.js

```diff
@@ -56,6 +59,25 @@ export default function(
   // 7. If the code matched by the syntactic production that is being evaluated is strict mode code, let strict be true, else let strict be false.
   let strict = strictCode;
 
-  // 8. Return a value of type Reference whose base value is bv, whose referenced name is propertyKey, and whose strict reference flag is strict.
-  return new Reference(bv, propertyKey, strict);
+  try {
+    // 8. Return a value of type Reference whose base value is bv, whose referenced name is propertyKey, and whose strict reference flag is strict.
+    return new Reference(bv, propertyKey, strict);
+  } catch (e) {
+    if (e instanceof FatalError && realm.isInPureScope() && bv instanceof Value && propertyKey instanceof Value) {
+      // havoc both values
+      Havoc.value(realm, bv);
+      Havoc.value(realm, propertyKey);
+      let val = AbstractValue.createTemporalFromBuildFunction(
+        realm,
+        Value,
+        [bv, propertyKey],
+        ([objNode, propertyNode]) => {
+          return t.memberExpression(objNode, propertyNode, ast.computed);
+        }
+      );
+      invariant(val instanceof AbstractValue);
+      return val;
```

Comment:
This is not safe for the case when this member expression will be used as a write instead of a read.

```js
foo[bar] = 123;
```

The generated code will try to treat this as a value. Probably generate something like `var $1 = foo[bar]; $1 = 123;` or something incorrect.

Reply:
We have checks for that elsewhere in the codebase:

https://github.com/facebook/prepack/blob/master/src/methods/properties.js#L975

Reply:
That results in a runtime error with a somewhat hard to understand error message. A compile time diagnostic would be better. Even better would be code to handle the actual assignment.